fvtest/compilerunittest/abstractinterpreter/AbsInterpreterTest.cpp

```diff
@@ -0,0 +1,443 @@
+}
+
+TEST_F(AbsVPValueTest, testMergeDiffferentParametersOperation) {
+    TR::AbsVPValue value1(vp(), NULL, TR::Int32);
+    value1.setParameterPosition(3);
+    TR::AbsVPValue value2(vp(), NULL, TR::Int32);
+    value2.setParameterPosition(4);
+    TR::AbsVPValue* mergedValue = static_cast<TR::AbsVPValue*>(value1.merge(&value2));
+    ASSERT_FALSE(mergedValue->isParameter());
+    ASSERT_EQ(-1, mergedValue->getParameterPosition());
+}
+
+class AbsOpStackTest : public TRTest::AbsInterpreterTest {};
+TEST_F(AbsOpStackTest, testSizeAndEmpty) {
+    TR::AbsOpStack stack(region());
+    ASSERT_EQ(0, stack.size());
+    ASSERT_TRUE(stack.empty());
+
+    stack.push(NULL);
+    ASSERT_EQ(1, stack.size());
+    ASSERT_FALSE(stack.empty());
+
+    stack.push(NULL);
+    ASSERT_EQ(2, stack.size());
+    ASSERT_FALSE(stack.empty());
+
+    stack.push(NULL);
+    ASSERT_EQ(3, stack.size());
+    ASSERT_FALSE(stack.empty());
+
+    stack.peek();
+    ASSERT_EQ(3, stack.size());
+    ASSERT_FALSE(stack.empty());
+
+    stack.pop();
+    ASSERT_EQ(2, stack.size());
+    ASSERT_FALSE(stack.empty());
+
+    stack.pop();
+    ASSERT_EQ(1, stack.size());
+    ASSERT_FALSE(stack.empty());
+
+    stack.pop();
+    ASSERT_EQ(0, stack.size());
+    ASSERT_TRUE(stack.empty());
+}
+
+TEST_F(AbsOpStackTest, testPushPopAndPeek) {
+    TR::AbsOpStack stack(region());
+
+    TRTest::AbsTestValue value1(TR::Int32, 0, 0);
+    TRTest::AbsTestValue value2(TR::Int32, 0, 0);
+    TRTest::AbsTestValue value3(TR::Int32, 0, 0);
+
+    // push and pop should correspond with each other
+    stack.push(&value1);
+    stack.push(&value2);
+    stack.push(&value3);
+    ASSERT_EQ(&value3, stack.peek());
+    ASSERT_EQ(&value3, stack.pop());
+    ASSERT_EQ(&value2, stack.peek());
+    stack.pop();
+    stack.pop();
+}
+
+TEST_F(AbsOpStackTest, testCloneOperation) {
+    TR::AbsOpStack stack1(region());
+
+    TRTest::AbsTestValue value1(TR::Int32, 0, 0);
+    TRTest::AbsTestValue value2(TR::Int32, 0, 0);
+    TRTest::AbsTestValue value3(TR::Int32, 0, 0);
+
+    stack1.push(&value1);
+    stack1.push(&value2);
+    stack1.push(&value3);
+
+    TR::AbsOpStack* stack2 = stack1.clone(region());
+
+    // cloned stack should have a different pointer than the original
+    ASSERT_NE(stack2, &stack1);
+
+    // clone does not affect stack1
+    ASSERT_EQ(3, stack1.size());
+    ASSERT_EQ(&value3, stack1.pop());
+    ASSERT_EQ(&value2, stack1.pop());
+    ASSERT_EQ(&value1, stack1.pop());
+
+    // cloned stack should have different pointers of abstract values.
+    ASSERT_EQ(3, stack2->size());
+
+    ASSERT_NE(&value3, stack2->pop());
+    ASSERT_NE(&value2, stack2->pop());
+    ASSERT_NE(&value1, stack2->pop());
+}
+
+TEST_F(AbsOpStackTest, testMergeOperation) {
+    TR::AbsOpStack stack1(region());
+
+    TRTest::AbsTestValue value1(TR::Int32, 0, 0);
+    TRTest::AbsTestValue value2(TR::Int32, 3, 5);
+    TRTest::AbsTestValue value3(TR::Int32,INT_MIN,INT_MAX);
```

Comment:
This call is still missing spaces after its commas:

```suggestion
    TRTest::AbsTestValue value3(TR::Int32, INT_MIN, INT_MAX);
```